packages/migrate/types/NodePath.ts

```diff
@@ -0,0 +1,89 @@
+export interface INode extends Object {
+	id?: {
+		name: string;
+	};
+	arguments?: INode[];
+	body?: INode[];
+	elements?: INode[];
+	expression?: {
+		left: {
+			computed: boolean,
+			object: INode,
+			property: INode,
+			type: string,
+		},
+		operator: string,
+		right: INode,
+		type: string,
+	};
+	filter?: (p: (p: INode) => boolean) => INode[] | any;
+	find?: (objectExpression: object, filterExpression?: object) => INode;
+	forEach?: (p: (p: INode) => void) => void;
+	get?: (property: string) => INode;
+	remove?: (_?: void) => void;
+	nodes?: (_?: void) => INode[];
+	pop?: (_?: void) => INode;
+	key?: {
+		name: string;
+		value: INode | string | any;
+	};
+	node?: INode;
+	name?: string;
+	object?: object;
+	parent?: INode;
+	properties?: INode[];
+	property?: INode;
+	prune?: Function;
+	replaceWith?: (objectExpression: object) => INode;
+	size?: (_?: void) => number;
+	type?: string;
+	value?: INode | string | any;
```

Comment:
same here

Reply:
I know. If we're keeping `INode | string`, then we'll have to do `typeof` check everywhere in the codebase. There are more than 100 places (like below) where we would have to do these checks. :disappointed: 

![image](https://user-images.githubusercontent.com/5961873/42633871-532796d4-85ff-11e8-8e71-2e31fad05407.png)

Reply:
Any chance that dynamic types could help in this case?

Reply:
@ematipico Can you please give an example how it would work in this case?

Reply:
I mean that maybe we have to intentionally use the `any` type because of the fact this particular case we want a soft typing... We are relying on the dynamic nature of JavaScript

Reply:
Yeah. I've done soft typing whenever we're using/assigning it. :100: